bears/vcs/git/GitCommitBear.py

```diff
@@ -204,3 +224,81 @@ def check_body(self, body,
             yield Result(self, 'Body of HEAD commit contains too long lines. '
                                'Commit body lines should not exceed {} '
                                'characters.'.format(body_line_length))
+
+    def check_issue(self, body,
+                    detect_host: bool=False,
+                    require_url: str=None,
+                    keyword_regexp: str=None):
+        """
+        Check for matching issue related URLs.
+
+        :param body:                Body of the commit message of HEAD.
+        :param detect_host:         Whether to automatically detect host
+                                    from git config or not.
+        :param require_url:         Whether the commit message requires
+                                    the presence of URL or not. 'full_ref'
+                                    confines commit to have full issue URL.
+                                    'short_ref' confines commit to have
+                                    shorter issue references like #123, etc.
+        :param keyword_regexp:      If provided, matches the keywords
+                                    with the regex. This gains precedence
+                                    over auto_detect_host and require_url.
+        """
+        if keyword_regexp:
+            if not re.search(keyword_regexp, body):
+                yield Result(self, 'No match found for the regular '
+                                   'expression provided: %s' % keyword_regexp)
+            return
+
+        if not any([detect_host, require_url]):
+            return
+
+        remotes, _ = run_shell_command('git config --get-regex url')
+        remotes = [url.split()[-1] for url in remotes.splitlines()]
+        host = None
+
+        for url in remotes:
+            netloc = urlparse(url)[2]
+            if 'git@' in url:
+                netloc = url[url.find('@')+1:url.find(':')]
+                host = netloc.split('.')[0]
+                continue
+            netloc = urlparse(url)[1]
+            host = netloc.split('.')[0]
```

Comment:
So, you're suggesting to pick the message from commit body and identify the host, then check for presence of the site's keywords? That looks awesome 😮